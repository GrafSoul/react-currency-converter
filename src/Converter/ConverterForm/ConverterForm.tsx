import React from 'react';

type CurrencySelectProps = {
    classes: any;
    textForConvert: string;
    isReady: boolean;
    changeText: (event: React.ChangeEvent<HTMLInputElement>) => void;
    handleSubmit: () => void;
};

const ConverterForm = ({
    classes,
    textForConvert,
    changeText,
    handleSubmit,
    isReady,
}: CurrencySelectProps) => {
    return (
        <div className={classes.converterForm}>
            <label htmlFor="converter" className={classes.labelConverter}>
                Введите текст по шаблону -{' '}
                <span className={classes.textTemplate}>15 usd in rub</span>, <br />
                указав сумму и любые из существующих валют
            </label>
            <input
                type="text"
                id="converter"
                name="converter"
                value={textForConvert}
                onChange={changeText}
                className={classes.inputConverter}
            />
            <button
                onClick={handleSubmit}
                className={[classes.btnConverter, isReady ? classes.disabled : ''].join(' ')}
                disabled={isReady}
                title="Получить данные"
            >
                Получить
            </button>
        </div>
    );
};

export default ConverterForm;
