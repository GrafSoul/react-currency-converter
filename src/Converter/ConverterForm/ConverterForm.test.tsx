/* eslint-disable testing-library/render-result-naming-convention */
import { render, screen } from '@testing-library/react';
import ConverterForm from './ConverterForm';

import classes from '../Converter.module.scss';

describe('ConverterForm component', () => {
    it('ConverterForm renders', () => {
        render(
            <ConverterForm
                classes={classes}
                textForConvert={''}
                isReady={false}
                changeText={() => {}}
                handleSubmit={() => {}}
            />,
        );
        expect(screen.getByText(/Введите текст по шаблону/i)).toBeInTheDocument();
    });

    it('ConverterForm snapshot', () => {
        const converterForm = render(
            <ConverterForm
                classes={classes}
                textForConvert={''}
                isReady={false}
                changeText={() => {}}
                handleSubmit={() => {}}
            />,
        );
        expect(converterForm).toMatchSnapshot();
    });
});
