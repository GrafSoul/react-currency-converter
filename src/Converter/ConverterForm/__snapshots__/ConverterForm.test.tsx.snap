// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converter component Converter snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="converterForm"
      >
        <label
          class="labelConverter"
          for="converter"
        >
          Введите текст по шаблону -
           
          <span
            class="textTemplate"
          >
            15 usd in rub
          </span>
          , 
          <br />
          указав сумму и любые из существующих валют
        </label>
        <input
          class="inputConverter"
          id="converter"
          name="converter"
          type="text"
          value=""
        />
        <button
          class="btnConverter "
          title="Получить данные"
        >
          Получить
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="converterForm"
    >
      <label
        class="labelConverter"
        for="converter"
      >
        Введите текст по шаблону -
         
        <span
          class="textTemplate"
        >
          15 usd in rub
        </span>
        , 
        <br />
        указав сумму и любые из существующих валют
      </label>
      <input
        class="inputConverter"
        id="converter"
        name="converter"
        type="text"
        value=""
      />
      <button
        class="btnConverter "
        title="Получить данные"
      >
        Получить
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ConverterForm component ConverterForm snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="converterForm"
      >
        <label
          class="labelConverter"
          for="converter"
        >
          Введите текст по шаблону -
           
          <span
            class="textTemplate"
          >
            15 usd in rub
          </span>
          , 
          <br />
          указав сумму и любые из существующих валют
        </label>
        <input
          class="inputConverter"
          id="converter"
          name="converter"
          type="text"
          value=""
        />
        <button
          class="btnConverter "
          title="Получить данные"
        >
          Получить
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="converterForm"
    >
      <label
        class="labelConverter"
        for="converter"
      >
        Введите текст по шаблону -
         
        <span
          class="textTemplate"
        >
          15 usd in rub
        </span>
        , 
        <br />
        указав сумму и любые из существующих валют
      </label>
      <input
        class="inputConverter"
        id="converter"
        name="converter"
        type="text"
        value=""
      />
      <button
        class="btnConverter "
        title="Получить данные"
      >
        Получить
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
